tests/test_parser.py

```diff
@@ -844,6 +844,58 @@ def test_rates():
     )  # Hasn't been transformed to "denominator" yet
 
 
+def test_ping_ordering():
+    contents = parser.parse_objects(
+        [ROOT / "data" / "pings.yaml"],
+        config={"allow_reserved": False},
+    )
+
+    errs = list(contents)
+    assert len(errs) == 0
+
+    pings = list(contents.value["pings"].keys())
+
+    assert pings == sorted(pings)
+
+
+def test_metric_ordering():
+    all_metrics = parser.parse_objects(
+        [ROOT / "data" / "ordering.yaml"], config={"allow_reserved": False}
+    )
+
+    errs = list(all_metrics)
+    assert len(errs) == 0
+
+    category = all_metrics.value["testing.ordering"]
+
+    assert len(category.values()) > 0
+    test_metrics = [f"{m.category}.{m.name}" for m in category.values()]
```

Comment:
Uh, but won't these already be sorted on input, since they're in order in the yaml? Maybe we need a `fourth_test_metric` which is lexicographically second?